crates/router/src/db/configs.rs

```diff
@@ -113,47 +114,110 @@ impl ConfigInterface for Store {
 impl ConfigInterface for MockDb {
     async fn insert_config(
         &self,
-        _config: storage::ConfigNew,
+        config: storage::ConfigNew,
     ) -> CustomResult<storage::Config, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let mut configs = self.configs.lock().await;
+
+        let config_new = storage::Config {
+            #[allow(clippy::as_conversions)]
+            id: configs.len() as i32,
+            key: config.key,
+            config: config.config,
+        };
+        configs.push(config_new.clone());
+        Ok(config_new)
     }
 
     async fn find_config_by_key(
         &self,
-        _key: &str,
+        key: &str,
     ) -> CustomResult<storage::Config, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let configs = self.configs.lock().await;
+        let config = configs.iter().find(|c| c.key == key).cloned();
+
+        match config {
+            Some(c) => Ok(c),
+            None => {
+                Err(errors::StorageError::ValueNotFound("cannot find config".to_string()).into())
+            }
+        }
```

Comment:
You could use `ok_or_else()` here and elsewhere.

Reply:
It should be done now! Also fixed the warning of clippy